mlflow/gateway/providers/mosaicml.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, Dict
+
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from mlflow.gateway.config import MosaicMLConfig, RouteConfig
+from mlflow.gateway.providers.base import BaseProvider
+from mlflow.gateway.providers.utils import rename_payload_keys, send_request
+from mlflow.gateway.schemas import chat, completions, embeddings
+
+
+class MosaicMLProvider(BaseProvider):
+    def __init__(self, config: RouteConfig) -> None:
+        super().__init__(config)
+        if config.model.config is None or not isinstance(config.model.config, MosaicMLConfig):
+            raise TypeError(f"Unexpected config type {config.model.config}")
+        self.mosaicml_config: MosaicMLConfig = config.model.config
+
+    async def _request(self, model: str, payload: Dict[str, Any]) -> Dict[str, Any]:
+        headers = {"Authorization": f"{self.mosaicml_config.mosaicml_api_key}"}
+        return await send_request(
+            headers=headers,
+            base_url=self.mosaicml_config.mosaicml_api_base
+            or "https://models.hosted-on.mosaicml.hosting",
+            path=model + "/v1/predict",
+            payload=payload,
+        )
+
+    async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
+        raise HTTPException(status_code=404, detail="The chat route is not available for MosaicML.")
```

Comment:
Why not add this now? We are doing a fast follow-up to add chat completions support for the databricks version as well. We should get this in time for mlflow 2.7.0.

(If you don't want to increase the scope of this PR, we can do it in a follow up.)

Reply:
See https://databricks.atlassian.net/browse/MLOPS-163

Reply:
Once we align on https://github.com/databricks/universe/pull/374719, we can add that support here as well

Reply:
Actually just ended up adding equivalent support here in the interest of time, and will update both as necessary PTAL